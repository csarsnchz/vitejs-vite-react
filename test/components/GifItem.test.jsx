import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';
describe('Pruebas en componente <GifItem />', () => {
   const title = 'Goku GIF by Dragon Ball Super';
   const url='https://media2.giphy.com/media/irBHYSZxbUifTxTgBL/giphy-downsized-medium.gif?cid=e38884cdh9dao9hezjmv3e31vsmxroboxvywl2qf9b8qwsvf&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g';
   test('Debe hacer match con la configuraciÃ³n de snapshot', () => {
     const { container } = render(<GifItem title={title} url={url}/>);
     expect( container ).toMatchSnapshot();
   });
   test('Debe mostrar la imagen con el URL y el ALT indicado',()=>{
      render(<GifItem title={title} url={url} />);
      const { src, alt } = screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(title);
   });
   test('Debe mostrar el titulo en el componente',()=>{
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText( title )).toBeTruthy();
   });
});