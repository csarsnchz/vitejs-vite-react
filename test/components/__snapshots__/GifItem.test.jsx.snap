// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifItem /> Debe hacer match con la configuraci√≥n de snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku GIF by Dragon Ball Super"
      src="https://media2.giphy.com/media/irBHYSZxbUifTxTgBL/giphy-downsized-medium.gif?cid=e38884cdh9dao9hezjmv3e31vsmxroboxvywl2qf9b8qwsvf&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g"
    />
    <p>
      Goku GIF by Dragon Ball Super
    </p>
  </div>
</div>
`;
